### ISSUE TITLE
Rename `easy_my_coop_website` to `cooperator_website`

### ISSUE THREAD

**Issue by polchampion** [coopiteasy/vertical-cooperative#214]
- Rename easy_my_coop_website to cooperator_website

**Comment by robinkeunen** [comment]
@remytms tu peux relire et compléter la spec ?

**Comment by victor-champonnois** [comment]
@robinkeunen Je vois un premier problème : website_recaptcha dépends de website.

**Comment by victor-champonnois** [comment]
@robinkeunen Note : website_recaptcha dépends de website, donc website sera installé dans tous les cas si on a besoin du captha. Est-ce que ça affecte la tâche ?

**Comment by robinkeunen** [comment]
@victor-champonnois I think you should remove the dependency to website in `website_recaptcha` as well. You will need to move the configurations introduced in `view_website_config_settings` elsewhere meaningful. Other than that, I do not think we need the dependency.

@remytms do you agree with this.

**Comment by victor-champonnois** [comment]
@polchampion After discussing with Robin and Rémy, we need help on the functional side for this task.
Basically, if we do this task, we will lose the top bar that permits someone going to the Odoo website and who is not logged in to click on the "Become Cooperator"  button. This means that the front end form will only be accessible by clicking on a link from another website (le site vitrine). 
The question is, do some of the clients need to have the front end interface for their odoo website from which someone who does not have an account can click on Become cooperator to fill the subscription form ?

**Comment by victor-champonnois** [comment]
@robinkeunen @remytms After discussing with @polchampion , the conclusion is not to do it.
- some client use the website module for other stuff than just accessing the form
- It would add more work to the analysts to determine whether a client is actively using website or not, and whether it should be uninstalled. 
- For these clients, the automated task to add the button on the top bar that is currently done by the module would instead be done by hand (mostly by the analyst), which is not desirable.

**Comment by robinkeunen** [comment]
Ok 👌 Thanks for the investigation !

**Comment by victor-champonnois** [comment]
Related PR:
1. For removing website dependency in website_recaptcha, and renaming to portal_recaptcha
- https://github.com/coopiteasy/vertical-cooperative/pull/272 (in vertical coop)
- https://github.com/coopiteasy/addons/pull/218 (in addons)

2. For renaming easy_my_coop_website to cooperator_website
- https://github.com/coopiteasy/vertical-cooperative/pull/292 (vertical coop)
- https://github.com/beescoop/Obeesdoo/pull/327 (obeesdoo)

**Comment by vdewulf** [comment]
@victor-champonnois I wanted to test this (I understood that's what you merged and deployed this morning). On which database can I test it (I want to make sure I'm testing the right code)?
And what is to be tested exactly? I understand you removed some dependencies but finally not all of them, so what would be the thing to be tested at the end?
Do we need to test this on a customer test database with existing data and/or on a new database to make sure that the installation process is going smoothly and verify which module is installed by installing other modules (in which order shoud we do it? cooperator > then website_cooperator > then website_recaptcha) ?
Thanks!

**Comment by victor-champonnois** [comment]
@vdewulf 
- You can test it on any komunigi databases (oufticoop, lenid, demain). 
- To be tested : that the captcha and the captcha settings works, and more broadly that all the website part of easy_my_coop (basically the subscription form) works. Also check that translations works for captcha settings and website form.
- I think testing on customer database is enough, since we have the CI to check that everything works (no errors and tests pass) at install on a fresh database.

**Comment by robinkeunen** [comment]
# State of the issue 2022-04-13

## Refactor website_recaptcha_reloaded

`website_recaptcha_reloaded` was renamed `website_recaptcha` and moved from _vertical-cooperative_ to *addons*

- https://github.com/coopiteasy/vertical-cooperative/pull/231
- https://github.com/coopiteasy/addons/pull/216
	- should point to banch `12.0-komunigi`
- https://github.com/beescoop/Obeesdoo/pull/313
	- should point to banch `12.0-komunigi`

`website_recaptcha` now depends on `portal` rather than `website` 

- https://github.com/coopiteasy/vertical-cooperative/pull/272
	- depends on #231
	- contains commits from #231 
	- points to 12.0-cooperator-xml-ids instead of #231 or `12.0-komunigi`
- https://github.com/coopiteasy/addons/pull/218
	- depends on #216

As discussed in https://github.com/coopiteasy/vertical-cooperative/issues/283 , an OCA module should not depend on a non-oca module in the long term.

### TODO :

- `website_recaptcha` (formely `website_recaptcha_reloaded`) should be named `portal_recaptcha`
    - https://github.com/coopiteasy/vertical-cooperative/issues/221
    -> Victor : done
- leave `portal_recaptcha` in _vertical-cooperative_
    -> Victor : done (in same issue as above)
- refactor recaptcha code out of `cooperator_website` into module `cooperator_portal_recaptcha`. Then later consider moving these modules to the OCA
    - https://github.com/coopiteasy/vertical-cooperative/issues/283

## Rename easy_my_coop_website to cooperator_website

- https://github.com/coopiteasy/vertical-cooperative/pull/292
- https://github.com/beescoop/Obeesdoo/pull/327

Dépendent du travail ci-dessus (surtout #231)

**Comment by robinkeunen** [comment]
@carmenbianca @victor-champonnois état du travail ici

**Comment by robinkeunen** [comment]
@vdewulf il va encore y avoir des changements substantiels sur cette tâche, pas la peine de tester tout de suite

**Comment by vdewulf** [comment]
ok, merci, j'attends votre GO
-----Original Message-----From: Robin Keunen ***@***.***>Reply-To: 
coopiteasy/vertical-cooperative <
***@***.***>To:
coopiteasy/vertical-cooperative ***@***.***>Cc:
vdewulf ***@***.***>, Mention ***@***.***>Subject:
Re: [coopiteasy/vertical-cooperative] Rename `easy_my_coop_website` to
`cooperator_website` (Issue #214)Date: Wed, 13 Apr 2022 10:11:20 -0700

@vdewulf il va encore y avoir des changements substantiels sur cette tâche, pas
la peine de tester tout de suite

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: <
***@***.***>

-- 
Virginie
0477/64.17.20

**Comment by robinkeunen** [comment]
@victor-champonnois @carmenbianca merge into `12.0-komunigi` in each repo (vertical-cooperative and obeesdoo) as soon as the CI is green and reviews are green. Let's simplify the deployments.

**Comment by carmenbianca** [comment]
@robinkeunen I was just going to suggest that. There is no starting here unless I merge a whole lot of stuff.
